sdk/identity/test-resources.bicep

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+@minLength(6)
+@maxLength(23)
+@description('The base resource name.')
+param baseName string = resourceGroup().name
+
+@description('Whether to deploy resources. When set to false, this file deploys nothing.')
+param deployResources bool = false
```

Comment:
With every resource conditioned on `deployResources`, what's the point in even defining the conditionality? Wouldn't it make more sense just not to run the script if you don't need the resources? I'm trying to understand why have the entire file conditioned and in what circumstances you'd pass `true` here.

Reply:
This is future-proofing. We'll eventually have live tests that don't need deployed resources. I'll want to run those tests in all internal pipelines, and the tests that need deployed resources only in the weekly pipeline. When I set all this up for Go, pipelines always deployed resources when `RunLiveTests: true`. Adding a conditional to the bicep resources that's true only in the weekly pipeline (see `managed-identity-matrix.json`) makes the deployment step a no-op in other pipelines. So far as I know, we still need a workaround like this; please let me know if you have other ideas